Static packaging type 'bundle' instead of property

mkarg opened the pull request:
Currently used maven-bundle-plugin 3.5.0 seems to work fine with module-info, so apparently there is no more need for separate packaging types on JDK 8- vs. JDK 9+.

Closing #627

mkarg commented:
Travis fails on JDK 9 due to missing JAXB, which is strange, as this works for me:

```
[WARNING] Tests run: 42, Failures: 0, Errors: 0, Skipped: 3
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 27.573 s
[INFO] Finished at: 2018-05-13T13:26:40+02:00
[INFO] ------------------------------------------------------------------------

Markus@Zauselmaus MINGW64 ~/git/jaxrs-api-master/jaxrs-api (mkarg-packaging-type)
$ mvn -version
Apache Maven 3.5.3 (3383c37e1f9e9b3bc3df5050c29c8aff9f295297; 2018-02-24T20:49:05+01:00)
Maven home: C:\Program Files\apache-maven-3.5.3
Java version: 9.0.4, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk-9.0.4
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 8.1", version: "6.3", arch: "amd64", family: "windows"
```
Any ideas?

chkal commented:
I just checked this and the build is working fine for me locally too.

But I think that I know what is causing this. The version of the `maven-surefire-plugin` used during the build is different between Travis (`2.21.0`) and my box (`2.20.1`). And there is an important difference between those versions.

From the [plugin page](https://maven.apache.org/surefire/maven-surefire-plugin/java9.html):
* Version 2.20.1 of the plugin added automatically “–add-modules java.se.ee” option to the command line of forked JVMs [...] in order to ease the transition of applications to Java 9.
* From 2.21 onwards the plugin will not add that option: the recommended way of using those modules is to add explicit dependencies of the maintained versions of such libraries.

We should really manage the version of the `maven-surefire-plugin` to get consistent builds between environments.

mkarg commented:
@chkal You're my here! I will change that.

mkarg commented:
The problem actually was not as expected (missing java.xml.bind is explicitly mentioned in our POM) but the fact that JAXB wants *reflective* access to JAX-RS. I fixed that in PR #630.

So I rebased *this* PR ontop of #630 and wait until #630 is merged first.

chkal commented:
@mkarg One question. My impression was that it is actually the failing test which triggered the reflection and therefore caused the failure. So maybe we don't need a new entry in the production JAR at all but just for the test? In this case I would vote for not adding this entry to the `module-info.java` but instead change the configuration of the `maven-surefire-plugin`. Disclaimer: I'm not very with Java modules, so maybe I'm totally wrong on this.

mkarg commented:
@chkal Yes, you are right! I now replaced general modification of `module-info` by particular opening to reflecting only for `maven-surefire-plugin`. Thanks for pointing this out!

mkarg commented:
@asoldano @spericas Please redo review as I had to adapt the solution: Only opening to reflection in surefire tests, not generally in module-info.

mkarg commented:
Merged after 13 days of review already (instead of the default 14 days) because no API is getting changed and the most active committers voted +1.
